front/lib/api/chat.ts

```diff
@@ -299,3 +327,335 @@ export async function updateChatMessageFeedback({
+ */
+export async function* newChat(
+  auth: Authenticator,
+  userMessage: string,
+  dataSources:
+    | {
+        workspace_id: string;
+        data_source_id: string;
+      }[]
+    | null,
+  filter: {
+    timestamp: { gt: number };
+  } | null,
+  timeZone: string
+): AsyncGenerator<
+  | ChatSessionCreateEvent
+  | ChatMessageTriggerEvent
+  | ChatMessageCreateEvent
+  | ChatMessageTokensEvent
+  | ChatSessionUpdateEvent
+> {
+  const owner = auth.workspace();
+  if (!owner) {
+    throw new Error("Could not authenticate against the workspace.");
+  }
+
+  // create a Session
+  const sId = newChatSessionId();
+  const session = await upsertChatSession(auth, sId, null);
+
+  if (!session) {
+    // If we fail to create the session we simply yield error messages events and return.
+    yield { role: "error" } as ChatMessageTriggerEvent;
+    yield {
+      message: {
+        sId: new_id(),
+        role: "error",
+        message: "Could not create chat session.",
+      },
+    };
+    return;
+  }
+
+  yield { session } as ChatSessionCreateEvent;
+
+  const assistantConfig = cloneBaseConfig(
+    DustProdActionRegistry["chat-assistant-wfn"].config
+  );
+
+  const assistantContext = {
+    user: {
+      username: auth.user()?.username,
+      full_name: auth.user()?.name,
+    },
+    workspace: owner.name,
+    date_today: new Date().toISOString().split("T")[0],
+  };
+
+  const messages: ChatMessageType[] = [
+    {
+      sId: new_id(),
+      role: "user",
+      message: userMessage,
+    },
+  ];
+
+  // This is an helper function that adds an error message to the `messages` array and yield the
+  // proper events in case of error.
+  function* handleError(
+    errorMessage: string
+  ): Generator<ChatMessageTriggerEvent | ChatMessageCreateEvent> {
+    messages.push({
+      sId: new_id(),
+      role: "error",
+      message: errorMessage,
+    });
+
+    yield { role: "error" } as ChatMessageTriggerEvent;
+    yield {
+      message: messages[messages.length - 1],
+    } as ChatMessageCreateEvent;
+  }
+
+  // Master loop that will run until the last message is not a "retrieval" request. It will call the
+  // assistant and push to the `messages` array. The assistant either push an "assistant" message
+  // with a response or a "retrieval" message that will contain a query, if that's the case we run
+  // the retrieval and add the retrieved documents to that last message and loop back here.
+  //
+  // At any point if an error occured an "error" message is pushed to the `messages` array and we
+  // will exit from this loop.
+  while (messages[messages.length - 1].role !== "retrieval") {
+    const res = await runActionStreamed(
+      auth,
+      "chat-assistant-wfn",
+      assistantConfig,
+      [{ messages: filterMessagesForModel(messages), assistantContext }]
+    );
+
+    if (res.isErr()) {
+      yield* handleError(
+        `Chat error: [${res.error.type}] ${res.error.message}`
+      );
+      break;
```

Comment:
at that stage the error message will be stored in DB (+ change title etc), cf the discussion we had IRL on the topic
This changes the existing logic; we can do it but warrants a new discussion & clarification in the code IMHO

Other option is to keep the existing logic by throwing. The caller of newChat can then decide how to handle the error. Or just exit the generator with return

Reply:
Let's simply not store error message for now :+1: 

We can't throw, we're streaming so the API endpoint won't have an opportuinity to retrun an error